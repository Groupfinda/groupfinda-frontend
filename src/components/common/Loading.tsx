import React from "react";
import { Spinner, Modal as Test } from "@ui-kitten/components";
import { StyleSheet, View, Modal } from "react-native";

type Props = {
  visible: boolean;
  status?:
    | "basic"
    | "primary"
    | "success"
    | "info"
    | "warning"
    | "danger"
    | "control";
  size?: "tiny" | "small" | "medium" | "large" | "giant";
};

const Loading: React.FC<Props> = (props) => {
  const { visible, status, size } = props;

  /*
  return (
    <View style={styles.container}>
      <Test backdropStyle={styles.backdrop} visible={true}>
        <Spinner status={status || "primary"} size={size || "medium"} />
      </Test>
    </View>
  );
  */
  return (
    <Modal visible={visible} transparent={true}>
      <View style={styles.backdrop}>
        <Spinner status={status || "primary"} size={size || "medium"} />
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  backdrop: {
    flex: 1,
    backgroundColor: "rgba(0,0,0,0.5)",
    alignItems: "center",
    justifyContent: "center",
  },
});

export default Loading;
