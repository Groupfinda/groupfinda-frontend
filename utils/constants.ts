export const eventCategories = [
  "Outdoors",
  "Adventure",
  "Technology",
  "Health",
  "Sports",
  "Fitness",
  "Learning",
  "Food & Drink",
  "Culture",
  "Music",
  "Social",
  "Games",
  "Arts",
  "Pets",
  "Hobbies",
  "Volunteer",
];
