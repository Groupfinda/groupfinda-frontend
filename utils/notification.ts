import Constants from 'expo-constants'
import * as Permissions from 'expo-permissions'
import { Platform } from "react-native";
import * as Notifications from 'expo-notifications'

export async function registerForPushNotificationsAsync() {
    let token;
    if (Constants.isDevice) {
        const { status: existingStatus } = await Permissions.getAsync(Permissions.NOTIFICATIONS);
        let finalStatus = existingStatus;
        if (existingStatus !== 'granted') {
            const { status } = await Permissions.askAsync(Permissions.NOTIFICATIONS);
            finalStatus = status;
        }

        if (finalStatus !== 'granted') {
            alert("Notifications allow us to send you updates on group matches and chat messages!")
            return
        }

        token = (await Notifications.getExpoPushTokenAsync()).data
        console.log(token)
    }

    if (Platform.OS === 'android') {
        Notifications.setNotificationChannelAsync('Groupfinda', {
            name: 'Groupfinda',
            importance: Notifications.AndroidImportance.MAX,
            vibrationPattern: [0, 250, 250, 250],
            lightColor: "#FF231F7C"
        })
    }

    return token
}

export type GroupfindaBody = {
    id: string;
    type: string;
}

export type AndroidNotificationData = GroupfindaBody
export type IOSNotificationData = {
    aps: {
        alert: {
            body: string;
            "launch-image": string;
            subtitle: string;
            title: string;
        }
        category: string;
        sound: string;
        "thread-id": string;
    }
    body: GroupfindaBody
}