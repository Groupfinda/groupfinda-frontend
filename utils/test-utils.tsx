import React from "react";
import { ApplicationProvider, IconRegistry } from "@ui-kitten/components";
import * as eva from "@eva-design/eva";
import { myTheme } from "../custom-theme";
import { EvaIconsPack } from "@ui-kitten/eva-icons";
import { MockedProvider, MockedResponse } from "@apollo/react-testing";
import { render, RenderAPI, RenderOptions } from "react-native-testing-library";

const AllProviders = (apolloMocks: MockedResponse[]) => ({
  children,
}: {
  children: JSX.Element;
}) => {
  return (
    <>
      <IconRegistry icons={EvaIconsPack} />
      <ApplicationProvider {...eva} theme={{ ...eva.light, ...myTheme }}>
        <MockedProvider mocks={apolloMocks} addTypename={false}>
          {children}
        </MockedProvider>
      </ApplicationProvider>
    </>
  );
};

type CustomRenderType = (
  component: React.ReactElement<any>,
  options?: RenderOptions,
  apolloProps?: MockedResponse[]
) => RenderAPI;

const customRender: CustomRenderType = (
  ui,
  options,
  apolloProps: MockedResponse[] = []
) => render(ui, { wrapper: AllProviders(apolloProps), ...options });

export * from "react-native-testing-library";
export { customRender as render };
