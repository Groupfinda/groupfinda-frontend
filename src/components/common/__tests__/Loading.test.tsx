import React from "react";
import { render, cleanup } from "../../../../utils/test-utils";
import Loading from "../Loading";

afterEach(cleanup);

test("renders UI correctly on visible true", () => {
  const tree = render(<Loading visible={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders UI correctly on visible false", () => {
  const tree = render(<Loading visible={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders UI correctly on status danger", () => {
  const tree = render(<Loading visible={true} status="danger" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders UI correctly on size large", () => {
  const tree = render(<Loading visible={true} size="large" />).toJSON();
  expect(tree).toMatchSnapshot();
});
