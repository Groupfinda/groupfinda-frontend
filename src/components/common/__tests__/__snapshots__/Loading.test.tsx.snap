// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI correctly on size large 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Modal
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.5)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Size {
            "height": 28,
            "width": 28,
          }
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "height": 28,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "rotate": "45deg",
                },
              ],
              "width": 28,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                undefined,
                Size {
                  "height": 14,
                  "width": 28,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "height": 28,
                  "transform": Array [
                    Object {
                      "rotate": "-165deg",
                    },
                  ],
                  "width": 28,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Size {
                      "height": 14,
                      "width": 28,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#008EED",
                        "borderRadius": 14,
                        "borderWidth": 3.9,
                        "height": 28,
                        "width": 28,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "bottom": 0,
              "height": 28,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "rotate": "45deg",
                },
              ],
              "width": 28,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "top": 14,
                },
                Size {
                  "height": 14,
                  "width": 28,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "height": 28,
                  "transform": Array [
                    Object {
                      "translateY": -14,
                    },
                    Object {
                      "rotate": "345deg",
                    },
                  ],
                  "width": 28,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Size {
                      "height": 14,
                      "width": 28,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#008EED",
                        "borderRadius": 14,
                        "borderWidth": 3.9,
                        "height": 28,
                        "width": 28,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`renders UI correctly on status danger 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Modal
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.5)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Size {
            "height": 24,
            "width": 24,
          }
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "height": 24,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "rotate": "45deg",
                },
              ],
              "width": 24,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                undefined,
                Size {
                  "height": 12,
                  "width": 24,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "height": 24,
                  "transform": Array [
                    Object {
                      "rotate": "-165deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Size {
                      "height": 12,
                      "width": 24,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#FF2B2B",
                        "borderRadius": 12,
                        "borderWidth": 3.4,
                        "height": 24,
                        "width": 24,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "bottom": 0,
              "height": 24,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "rotate": "45deg",
                },
              ],
              "width": 24,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "top": 12,
                },
                Size {
                  "height": 12,
                  "width": 24,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "height": 24,
                  "transform": Array [
                    Object {
                      "translateY": -12,
                    },
                    Object {
                      "rotate": "345deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Size {
                      "height": 12,
                      "width": 24,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#FF2B2B",
                        "borderRadius": 12,
                        "borderWidth": 3.4,
                        "height": 24,
                        "width": 24,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`renders UI correctly on visible false 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Modal
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.5)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Size {
            "height": 24,
            "width": 24,
          }
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "height": 24,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "rotate": "45deg",
                },
              ],
              "width": 24,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                undefined,
                Size {
                  "height": 12,
                  "width": 24,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "height": 24,
                  "transform": Array [
                    Object {
                      "rotate": "-165deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Size {
                      "height": 12,
                      "width": 24,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#008EED",
                        "borderRadius": 12,
                        "borderWidth": 3.4,
                        "height": 24,
                        "width": 24,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "bottom": 0,
              "height": 24,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "rotate": "45deg",
                },
              ],
              "width": 24,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "top": 12,
                },
                Size {
                  "height": 12,
                  "width": 24,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "height": 24,
                  "transform": Array [
                    Object {
                      "translateY": -12,
                    },
                    Object {
                      "rotate": "345deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Size {
                      "height": 12,
                      "width": 24,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#008EED",
                        "borderRadius": 12,
                        "borderWidth": 3.4,
                        "height": 24,
                        "width": 24,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`renders UI correctly on visible true 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Modal
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.5)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Size {
            "height": 24,
            "width": 24,
          }
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "height": 24,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "rotate": "45deg",
                },
              ],
              "width": 24,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                undefined,
                Size {
                  "height": 12,
                  "width": 24,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "height": 24,
                  "transform": Array [
                    Object {
                      "rotate": "-165deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Size {
                      "height": 12,
                      "width": 24,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#008EED",
                        "borderRadius": 12,
                        "borderWidth": 3.4,
                        "height": 24,
                        "width": 24,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "bottom": 0,
              "height": 24,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "rotate": "45deg",
                },
              ],
              "width": 24,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "top": 12,
                },
                Size {
                  "height": 12,
                  "width": 24,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "height": 24,
                  "transform": Array [
                    Object {
                      "translateY": -12,
                    },
                    Object {
                      "rotate": "345deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Size {
                      "height": 12,
                      "width": 24,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#008EED",
                        "borderRadius": 12,
                        "borderWidth": 3.4,
                        "height": 24,
                        "width": 24,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
